src/pkynetics/model_fitting_methods/coats_redfern.py

```
@@ -79,9 +86,25 @@
+        )
+        # Use absolute value to proceed with calculation while avoiding math errors
+        e_a_for_calc = abs(e_a)
+    else:
+        e_a_for_calc = e_a
+
+    # Safely calculate pre-exponential factor
+    try:
+        ln_a = intercept + np.log(heating_rate / e_a_for_calc)
+        a = np.exp(ln_a)
+    except (ValueError, RuntimeWarning, FloatingPointError) as e:
+        logger.error(f"Error calculating pre-exponential factor: {str(e)}")
```

Ruff (0.8.2)

Undefined name `logger`

(F821)